packages/plugins/graphql-jit/src/index.ts

```diff
@@ -89,72 +83,77 @@ export const useGraphQlJit = (
       }
 
       jitCacheByDocument.set(args.document, cacheEntry);
-      if (documentSource) {
-        jitCacheByDocumentString.set(documentSource, cacheEntry);
+      if (jitCacheByDocumentString) {
+        const documentSource = getDocumentString(args.document);
+        if (documentSource) {
+          jitCacheByDocumentString.set(documentSource, cacheEntry);
+        }
       }
     }
     return cacheEntry;
   }
 
+  function jitExecutor(args: TypedExecutionArgs<unknown>) {
+    const cacheEntry = getCacheEntry(args);
+
+    const result$ = cacheEntry.subscribe
+      ? (cacheEntry.subscribe(
+          args.rootValue,
+          args.contextValue,
+          args.variableValues,
+        ) as PromiseOrValue<ExecutionResultWithSerializer>)
+      : (cacheEntry.query(
+          args.rootValue,
+          args.contextValue,
+          args.variableValues,
+        ) as PromiseOrValue<ExecutionResultWithSerializer>);
+    if (isPromise(result$)) {
+      return result$.then(r => {
+        r.stringify = cacheEntry.stringify;
+        return r;
+      });
```

Comment:
You use `isPromise` instead of `await` to avoid wrapping the result in a Promise if not necessary ?

Reply:
Extra async/await affects the performance somehow in Node.js :/